static/js/public/formValidation.js

```diff
@@ -0,0 +1,32 @@
+function formValidation() {
+  const form = document.querySelector(".marketo-form");
+  if (form) {
+    initFormValidation(form);
+  }
+}
+
+function initFormValidation(form) {
+  var startTime = Date.now();
+  var delta = null;
+
+  var timer = setInterval(function() {
+    delta = Date.now() - startTime; // milliseconds elapsed since start
+    // if the form is submitted after more than 3 seconds, it means it was not a bot
+    if (delta / 1000 > 3) {
+      spamCheck.value = "noBot";
+      clearInterval(timer);
+    }
+  }, 100); // update about every 100 miliseconds
+
+  var spamCheck = document.createElement("input");
+  spamCheck.id = "grecaptcharesponse";
+  spamCheck.type = "hidden";
+
+  const inputNode = form.querySelector("input[name=canonicalUpdatesOptIn]")
```

Comment:
Any reason this can't be appended straight into the form?

```
form.appendChild(spamCheck);
```

We already know the form exists. If for some reason the form exists but the "canonicalUpdatesOptIn" element doesn't this script won't work.